:bomb: `Bug`: `gitlab-ce-docker` update error

### What Happened?

apply twice

### How to Reproduce?

1. config:

```yaml
---
# core config
varFile: "" # If not empty, use the specified external variables config file
toolFile: "" # If not empty, use the specified external tools config file
pluginDir: ""
state:
  backend: local
  options:
    stateFile: devstream.state

---
tools:
# name of the tool
- name: gitlab-ce-docker
  # id of the tool instance
  instanceID: default
  # format: name.instanceID; If specified, dtm will make sure the dependency is applied first before handling this tool.
  dependsOn: [ ]
  # options for the plugin
  options:
    # hostname for running docker. (default: gitlab.example.com)
    hostname: gitlab.example.com
    # pointing to the directory where the configuration, logs, and data files will reside.
    # (default: /srv/gitlab)
    # 1. it should be a absolute path
    # 2. once the tool is installed, it can't be changed
    gitlab_home: /Users/danielhu/dtm-test/gitlab
    # ssh port exposed in the host machine. (default: 22)
    ssh_port: 22
    # http port exposed in the host machine. (default: 80)
    http_port: 8080
    # https port exposed in the host machine.
    # (default: 443)
    # todo: support https, reference: https://docs.gitlab.com/omnibus/settings/nginx.html#enable-https
    https_port: 443
    # whether to delete the gitlab_home directory when the tool is removed. (default: false)
    rm_data_after_delete: false
    # gitlab-ce tag. (default: "rc")
    image_tag: "rc"
```

2. apply

3. change config(e.g. change port from 80 to 8080)

4. apply

5. got logs

```sh
2022-09-01 16:46:38 ℹ [INFO]  Apply started.
2022-09-01 16:46:38 ℹ [INFO]  Using dir </Users/danielhu/.devstream/plugins> to store plugins.
2022-09-01 16:46:38 ℹ [INFO]  Using local backend. State file: devstream.state.
2022-09-01 16:46:38 ℹ [INFO]  Tool (gitlab-ce-docker/default) config drifted from the state, will be updated.
Continue? [y/n]
Enter a value (Default is n): y

2022-09-01 16:46:39 ℹ [INFO]  Start executing the plan.
2022-09-01 16:46:39 ℹ [INFO]  Changes count: 1.
2022-09-01 16:46:39 ℹ [INFO]  -------------------- [  Processing progress: 1/1.  ] --------------------
2022-09-01 16:46:39 ℹ [INFO]  Processing: (gitlab-ce-docker/default) -> Update ...
2022-09-01 16:46:39 !! [ERROR]  Options error: Key: 'Options.ImageName' Error:Field validation for 'ImageName' failed on the 'required' tag.
2022-09-01 16:46:39 !! [ERROR]  Options error: Key: 'Options.ImageTag' Error:Field validation for 'ImageTag' failed on the 'required' tag.
2022-09-01 16:46:39 !! [ERROR]  Options error: Key: 'Options.ContainerName' Error:Field validation for 'ContainerName' failed on the 'required' tag.
2022-09-01 16:46:39 ℹ [INFO]  -------------------- [  Processing aborted.  ] --------------------
2022-09-01 16:46:39 !! [ERROR]  Errors Map: key(gitlab-ce-docker/default-Update) -> value(opts are illegal)
2022-09-01 16:46:39 !! [ERROR]  Apply failed => some error(s) occurred during plugins apply process.
```

### Anything else

_No response_

### DevStream Version

latest

--- a/internal/pkg/plugin/gitlabcedocker/update.go
+++ b/internal/pkg/plugin/gitlabcedocker/update.go
@@ -31,7 +31,11 @@ func Update(options map[string]interface{}) (map[string]interface{}, error) {
 	}
 
 	// 3. update and get status
-	status, err := operator.Execute(options)
+	rawOptions, err := buildDockerOptions(opts).Encode()
+	if err != nil {
+		return nil, err
+	}
+	status, err := operator.Execute(rawOptions)
 	if err != nil {
 		return nil, err
 	}

